WordPress/Classes/ViewRelated/Reader/Headers/ReaderSiteHeaderView.swift

```diff
@@ -36,41 +40,13 @@ class ReaderSiteHeaderView: UITableViewHeaderFooterView, ReaderStreamHeader {
         headerViewModel.followerCount = siteTopic.subscriberCount.doubleValue.abbreviatedString()
         headerViewModel.isFollowingSite = siteTopic.following
     }
-
-    private func setupHeader() {
-        weak var weakSelf = self
-        let header = ReaderSiteHeader(viewModel: weakSelf?.headerViewModel ?? ReaderSiteHeaderViewModel())
-        let view = UIView.embedSwiftUIView(header)
-        addSubview(view)
-        NSLayoutConstraint.activate([
-            view.topAnchor.constraint(equalTo: topAnchor),
-            view.bottomAnchor.constraint(equalTo: bottomAnchor),
-            view.leadingAnchor.constraint(equalTo: readableContentGuide.leadingAnchor),
-            view.trailingAnchor.constraint(equalTo: readableContentGuide.trailingAnchor)
-        ])
-
-        addBottomBorder(withColor: .separator)
-    }
-
-    /// Sets the background color of the container view.
-    ///
-    /// For `UITableViewHeaderFooterView`, we'll need to assign a `UIView` with the desired
-    /// background color to the `backgroundView` property. Setting the `backgroundColor`
-    /// directly will pop a warning in the console.
-    ///
-    /// - Parameter color: The background color
-    private func applyBackgroundColor(_ color: UIColor) {
-        let backgroundView = UIView(frame: bounds)
-        backgroundView.backgroundColor = color
-        self.backgroundView = backgroundView
-    }
 }
 
 // MARK: - ReaderSiteHeader
 
-struct ReaderSiteHeader: View {
+private struct ReaderSiteHeader: View {
 
-    @StateObject var viewModel: ReaderSiteHeaderViewModel
+    @ObservedObject var viewModel: ReaderSiteHeaderViewModel
```

Comment:
Any particular reason for this change?

Reply:
`@StateObject` creates an autuclosure forcing you to use `self` here:

```swift
ReaderSiteHeader(viewModel: self.headerViewModel)
```

It's a potential retain cycle.

In this scenario, `@ObservedObject` is an obvious choice because `ReaderSiteHeaderView` manages the view model lifecycle. I'm not sure why it used to use `@StateObject`.